### PR TITLE
Add geometry check for map arithmetics

### PR DESCRIPTION
**Pull request by registerrier**
This PR adds a check on the `MapGeom` consistencies for each operation as well as the relevant tests.

A test for arithmetics after serialization is partly commented out as long as issue #1887 is still there.

### CODE REVIEW

File: gammapy/maps/base.py [hunk 1]
```diff
@@ -1036,8 +1036,10 @@ def __repr__(self):
     def _arithmetics(self, operator, other, copy):
         """ Perform arithmetics on maps after checking geometry consistency"""
         if isinstance(other, Map):
-            # TODO: check consistency
-            q = other.quantity
+            if self.geom == other.geom:
+                q = other.quantity
+            else:
+                raise ValueError("Map Arithemtics: Inconsistent geometries.")
```
Comment (cdeil) [inline gammapy/maps/base.py]: Typo in the word "arithmetics".

File: gammapy/maps/tests/test_base.py [hunk 2]
```diff
@@ -373,3 +373,40 @@ def test_map_arithmetics(map_type):
     m1 += 4
     assert m1.unit == u.Unit("")
     assert_allclose(m1.data, 4)
+
+
+def test_arithmetics_inconsistent_geom():
+    m_wcs = Map.create(binsz=0.1, width=1.0, map_type="wcs", skydir=(0, 0), unit="m2")
+    m_wcs_incorrect = Map.create(
+        binsz=0.1, width=2.0, map_type="wcs", skydir=(0, 0), unit="m2"
```
Comment (cdeil) [inline gammapy/maps/tests/test_base.py]: Maybe remove some arguments here that are the default anyways, then the line break will be gone and overall a bit shorter and less code to read in the tests? (e.g. `skydir=(0, 0)` is the default, `map_type="wcs"` as well).

File: gammapy/maps/base.py (hunk 1 above)
Comment (registerrier) [reply gammapy/maps/base.py]: Corrected

File: gammapy/maps/tests/test_base.py (hunk 2 above)
Comment (registerrier) [reply gammapy/maps/tests/test_base.py]: Done

**Comment by registerrier** [comment]
Changes done. Merging, now.

**Pull request merged**